crates/bevy_app/src/app.rs

```diff
@@ -229,6 +229,23 @@ impl App {
     }
 
     pub fn add_system_set(&mut self, set: SystemSet) -> &mut Self {
+        fn ensure_no_conflicts(set: &SystemSet) {
```

Comment:
I think this could keep a reference to the conflicting stage name and add that to the panic message.

Small suggestion, you could use this instead `conflict = set.systems.iter().any(|system| system.config().stage != set.config().stage);`, but that's entirely up to preference.